### PR TITLE
QuickInfoBar molecule is created and used inSideBar Organism

### PR DESCRIPTION
**Pull request by developervinit**
1. created Segment atom.
2. created QuickInfoBar molecule
3. used QuickInfoBar in SideBar organism
4. corrected some css.

QuickInfoBar image

### CODE REVIEW

File: apps/pm-user/src/components/molecules/quickInfoBar/QuickInfoBar.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      block={true}
+      defaultValue="received"
+      onChange={handleRedirectOnClick}
+      options={SEGMENT_OPTIONS}
+    />
+    </div>
+  );
+};
+
+export default QuickInfoBar;
```
Comment (mrkarthik3) [inline apps/pm-user/src/components/molecules/quickInfoBar/QuickInfoBar.js:36]: Why are you not taking props in QuickInfoBar component ?
Our end goal is to make this component reusable... right?
Then we must take props.

File: apps/pm-user/src/components/molecules/quickInfoBar/QuickInfoBar.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  function handleRedirectOnClick(value) {
+    // switch (value) {
+    //     case "received" :
+    //         return history.pushs("/recieved");
+    //     case "sent" :
+    //         return history.pushs("/sent");
+    //     case "accepted" :
+    //         return history.pushs("/accepted");
+    //     default :
+    //         return history.pushs("/recieved");
```
Comment (mrkarthik3) [inline apps/pm-user/src/components/molecules/quickInfoBar/QuickInfoBar.js:20]: Can you please explain this code ?

I understood that you are trying to redirect to different pages.

But the function (as far as I can remember) is `history.push()`.... but not `history.pushs()`

Also, there is no need to return it in my opinion. Just run` history.push("/yourtargetpage")`

And... the most important point I want to mention is....

We only have ONE mailbox page where the user can view all interests (Received/Sent/Rejected.....)
So, why are you thinking of multiple pages ? This is not what we agreed upon... right?

File: apps/pm-user/src/components/molecules/quickInfoBar/QuickInfoBar.js (hunk 1 above)
Comment (developervinit) [reply apps/pm-user/src/components/molecules/quickInfoBar/QuickInfoBar.js:36]: No it only one time use thing and it not having any props here.

File: apps/pm-user/src/components/molecules/quickInfoBar/QuickInfoBar.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    // }
+  }
+
+  return (
+    <div className={styles.quickInfoBar}>
+    <Segmented
+      block={true}
+      defaultValue="received"
+      onChange={handleRedirectOnClick}
```
Comment (mrkarthik3) [inline apps/pm-user/src/components/molecules/quickInfoBar/QuickInfoBar.js:29]: handleRedirectOnClick expects an argument as per your function definition.

if you do it this way, then only "default" case's code in switch statement will be reached. 

So, can you please explain what you are trying to do here ?

File: apps/pm-user/src/components/molecules/quickInfoBar/quickInfoBar.module.scss [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  & p{
+    font-weight: 600;
+    color: #5c5fee;
+  }
+  :global {
+    .ant-segmented {
+      background: transparent;
+    }
+    .ant-segmented-item-label {
+      padding: 5px;
```
Comment (mrkarthik3) [inline apps/pm-user/src/components/molecules/quickInfoBar/quickInfoBar.module.scss:11]: Using multiples of 4 is better.

File: apps/pm-user/src/components/molecules/quickInfoBar/QuickInfoBar.js (hunk 2 above)
Comment (developervinit) [reply apps/pm-user/src/components/molecules/quickInfoBar/QuickInfoBar.js:20]: history.pushs() will be corrected. and it is not for the "Mailbox" page it for the "Matches" page. dont be confuse it with messages

File: apps/pm-user/src/components/organisms/sideBar/SideBar.js [hunk 5]
```diff
@@ -1,28 +1,35 @@
...
+        />
+      </div>
+      <div className={styles.quickInfoBarWraper}>
+        <QuickInfoBar />
       </div>
       <div className={styles.menu}>
         <Menu
           defaultSelectedKeys={[DEFAULT_SELECTED_ITEM_KEY]}
           items={MENU_ITEMS}
           onClick={handleMenuItemClick}
```
Comment (mrkarthik3) [inline apps/pm-user/src/components/organisms/sideBar/SideBar.js:30]: is handleMenuItemClick required ?
If yes, then there is no code inside it... (lines 9-11).

So, is this onClick necessary ?

File: apps/pm-user/src/components/organisms/sideBar/SideBar.js [hunk 6]
```diff
@@ -1,28 +1,35 @@
...
+import { ProfileSummary } from '../../molecules';
+import { QuickInfoBar } from '../../molecules';
 import styles from './sideBar.module.scss';
-import { DEFAULT_SELECTED_ITEM_KEY, MENU_ITEMS } from './constants';
 
 function SideBar() {
-
   const handleMenuItemClick = ({ key }) => {
     //some code here
   };
```
Comment (mrkarthik3) [inline apps/pm-user/src/components/organisms/sideBar/SideBar.js:11]: Is this function even required ?

Comment (developervinit) [reply apps/pm-user/src/components/organisms/sideBar/SideBar.js:11]: this is created by Chitanaya not me

File: apps/pm-user/src/components/organisms/sideBar/SideBar.js (hunk 5 above)
Comment (developervinit) [reply apps/pm-user/src/components/organisms/sideBar/SideBar.js:30]: it is Chaitnay's code, not mine. please ask him

File: apps/pm-user/src/components/molecules/quickInfoBar/quickInfoBar.module.scss (hunk 4 above)
Comment (developervinit) [reply apps/pm-user/src/components/molecules/quickInfoBar/quickInfoBar.module.scss:11]: can elaborate it?

Comment (mrkarthik3) [reply apps/pm-user/src/components/molecules/quickInfoBar/quickInfoBar.module.scss:11]: Chaitanya suggested that it is a good practice to use multiples of 4 for all CSS properties.

Comment (developervinit) [reply apps/pm-user/src/components/molecules/quickInfoBar/quickInfoBar.module.scss:11]: ok I dont know it. you can add in it next PR

File: apps/pm-user/src/components/molecules/quickInfoBar/QuickInfoBar.js (hunk 3 above)
Comment (developervinit) [reply apps/pm-user/src/components/molecules/quickInfoBar/QuickInfoBar.js:29]: yes correcting it

File: apps/pm-user/src/components/organisms/sideBar/SideBar.js (hunk 6 above)
Comment (mrkarthik3) [reply apps/pm-user/src/components/organisms/sideBar/SideBar.js:11]: Ok... understood... Chaitanya wanted us to figure out how to redirect...
Use my code below... and update the function...
Then links will work...
Right ow all links are not working....
image

File: apps/pm-user/src/components/molecules/quickInfoBar/QuickInfoBar.js (hunk 2 above)
Comment (mrkarthik3) [reply apps/pm-user/src/components/molecules/quickInfoBar/QuickInfoBar.js:20]: Ok. understood.

File: apps/pm-user/src/components/molecules/quickInfoBar/QuickInfoBar.js (hunk 1 above)
Comment (mrkarthik3) [reply apps/pm-user/src/components/molecules/quickInfoBar/QuickInfoBar.js:36]: But for different use we have to show different data.... So, sideBar component should take props....

Comment (mrkarthik3) [reply apps/pm-user/src/components/molecules/quickInfoBar/QuickInfoBar.js:36]: either sideBar or QuickInfoBar should take props...

**Pull request merged**